stdlib/lazy.mli

```diff
@@ -67,36 +66,76 @@ external force : 'a t -> 'a = "%lazy_force"
    recursively.
 *)
 
-val force_val : 'a t -> 'a
-(** [force_val x] forces the suspension [x] and returns its
-    result.  If [x] has already been forced, [force_val x]
-    returns the same value again without recomputing it.
 
-    If the computation of [x] raises an exception, it is unspecified
-    whether [force_val x] raises the same exception or {!Undefined}.
-    @raise Undefined if the forcing of [x] tries to force [x] itself
-    recursively.
+(** {1 Iterators} **)
+
+val map : ('a -> 'b) -> 'a t -> 'b t
+(** [map f x] returns a suspension that, when forced,
+    forces [x] and applies [f] to its value.
+
+    It is equivalent to [lazy (f (Lazy.force x))].
+
+    @since 4.13.0
 *)
 
+(** {1 Reasoning on already-forced suspensions} *)
+
+val is_val : 'a t -> bool
+(** [is_val x] returns [true] if [x] has already been forced and
+    did not raise an exception.
+    @since 4.00.0 *)
+
+val from_val : 'a -> 'a t
+(** [from_val v] evaluates [v] first (as any function would) and returns
+    an already-forced suspension of its result.
+    It is the same as [let x = v in lazy x], but use dynamic tests
+    to optimize suspension creation in some cases.
+    @since 4.00.0 *)
+
+val map_val : ('a -> 'b) -> 'a t -> 'b t
+(** [map_val f x] applies [f] directly if [x] is already forced,
+   otherwise it behaves as [map f x].
+
+   When [x] is already forced, this behavior saves the construction of
+   a suspension, but on the other hand it perform more work eagerly
+   that may not be useful if you never force the function result.
```

Comment:
```suggestion
   that may not be useful if you never force the function's result.
```

Reply:
This is a nitpick, but I think that "the function result" is correct (at least it is commonly used) and I find "the function's result" too heavy, so I vote to discard this change.